src/components/Configure/state/utils.ts

```diff
@@ -27,12 +26,30 @@ import {
   isIntegrationFieldMapping,
 } from "../utils";
 
-// uses lodash deep equality check to compare two saved write objects (typed checked)
+// compares two write objects by checking if they have the same objectNames
+// implementation detail: sdk generates selectedValueDefaults, selectedFieldSettings;
+// todo: switch to not pass the undefined values to the sdk
```

Comment:
this sentence is confusing, I think what you mean is "update openapi spec so the generated SDK no longer have undefined values for selectedValueDefaults, selectedFieldSettings"

nothing is passing undefined values to the SDK, the SDK is generating fields with undefined values